### DISCUSSION TITLE
[Ethics & legality] Mapbox V10 is not free, so why invest time to making Mapbox money while you get nothing?

### DISCUSSION THREAD

**Discussion by aliciaschoen** [rnmapbox/maps#2471] [General]
I'm wondering what's the motivation for the maintainers to spend their time on earth building a software so another company can get more customers and make more money when they can support other open source projects than can do almost the same thing? 

It made sense until Mapbox moved off from the open source license. Or is Mapbox paying the contributors? 

Also, is it even legal to Modify or or build on top of Mapbox's SDK code and redistribute it? All it might take is this code breaking on a big platform giving Mapbox a bad image before they come after this repository and shut it down. In such case, would it be legal to continue this repository?

I'm atm deciding which Library to use to integrate maps into my Platform and Mapbox V10 looks like the BEST choice with lots of awesome features, but I'm very skeptical with the longevity of the repository, looking  5 years onward!

Summed up; what's the motivation to maintain the code and what's the legality around it?

Thanks

**Comment by systemlevel** [comment]
Hi @aliciaschoen, many projects gain value from what Mapbox is offering. If that is something you need in your project, it makes sense to use Mapbox in those cases.  Their newest releases include a lot of value add. Each project will need to evaluate whether or not that is needed.

**Comment by systemlevel** [comment]
Mapbox is not paying the contributors, to my knowledge. It's community supported project at this time. We have been contributing by supporting some of the main contributors to the repository and are looking to expand our support if any additional contributors would like to be sponsored for their work.

**Comment by KiwiKilian** [comment]
My current employer has a project using React Native and the need to use Mapbox. The current project is committed to use MapBox as the web app of the same project already uses it `¯\_(ツ)_/¯`. Therefore I'm a contributing every now and then where features/fixes are needed for our project. We are currently also exploring possibilities to switch towards MapLibre (mainly on the web). I totally understand your concerns and I'm also bothered by this topic. Regarding this I recommend to follow the [process over at MapLibre](https://github.com/maplibre/maplibre-react-native).

As in any case with open source and software in general, there are never any guarantees how long a pieace of software will be supported or work.

**Reply by RyanTG** [reply]
Is there a fair amount of feature parity between the maplibre-react-native repo and this one? I'm having trouble figuring that out.

We currently use react-native-maps, but I'm not super happy with it. We host our tile server, but cannot risk the completely unknown (to us) costs of mapbox. It's not possible to use this repo and exclude the `MapboxGL.setAccessToken('<YOUR_ACCESSTOKEN>');` portion, right? I'm just trying to figure out our options (maplibre-react-native or this repo).

**Reply by KiwiKilian** [reply]
It is not possible to use a current Mapbox SDK without their tokens. For using MapLibre one should definitely [use their fork](https://github.com/maplibre/maplibre-react-native) which was currently released.

Otherwise one has to compare the functionality of the native SDKs from MapLibre and Mapbox. I didn't dig deeply into this, but I guess for most use cases, MapLibre does a solid job. Maybe Mapbox is a few features ahead, but really depends on what is needed. There is also very interesting stuff going on at MapLibre, thinking about rebuilding with Metal on iOS.
